### PR TITLE
[do not merge] Edits to procedures in the dev_guide/builds file

### PR DESCRIPTION
**Pull request by bfallonf**
This is an edits to the procedures in the dev_guide/buildes file, originally in PR #443 , but the conflicts got too much (I'll have to close that one). @soltysh I can see you incorporated your feedback from my questions from that PR into your edits. Big thanks!

However, this does mean your edits to this file in the #492 might be a bit...conflict-ey. If we're done with the builds file it'd be so great to this PR merged. My original PR was created 11 days ago and I need to stop playing with conflicts here...

@mfojtik @bparees , tagging you guys as well, in case there's any other edits to the builds file I need to be worried about...

### CODE REVIEW

**Comment by bparees** [comment]
i'm staying out of these files until the dust settles.  but please do let me know when that has happened :)

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -80,22 +80,29 @@ every time a Docker image tag or the source code changes:
...
-in strategy definition as presented below:
-
-====
+Source-To-Image (S2I) builds provide an option to perform incremental builds,
+meaning they will reuse artifacts from previously built images. To create an
+incremental build create a BuildConfig much like the above example, but
+with a modification to the strategy parameter:
 
-[source,json]
+[incremental.json]
```
Comment (adellape) [inline dev_guide/builds.adoc]: This won't actually render/do anything.

**Comment by bfallonf** [comment]
@soltysh @mfojtik @bparees Realised this PR has outdated information in it, so it's not worth merging. I'll wait for you all to be done with the builds stuff then I'll probably start again in another PR.

**Comment by soltysh** [comment]
@bfallonf sorry, that wasn't intended :smile_cat:

**Comment by bfallonf** [comment]
Edits continued here: https://github.com/openshift/openshift-docs/pull/587

**Pull request closed without merging**